synapse/storage/data_stores/main/schema/delta/58/09unread_messages.sql

```diff
@@ -0,0 +1,22 @@
+/* Copyright 2020 The Matrix.org Foundation C.I.C
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+CREATE TABLE IF NOT EXISTS unread_messages(
+    user_id TEXT NOT NULL,  -- The user for which the message is unread.
+    stream_ordering BIGINT NOT NULL,  -- The position of the message in the event stream.
+    room_id TEXT NOT NULL,
+    event_id TEXT NOT NULL,  -- The ID of the message, we need it to handle redactions.
```

Comment:
per #synapse-dev: I'm wondering  if we should have foreign key constraints here.